rust/src/treefile.rs

```diff
@@ -1261,63 +1265,62 @@ etc-group-members:
     }
 
     #[test]
-    fn test_split_whitespace_unless_quoted() {
+    fn test_split_whitespace_unless_quoted() -> Result<()> {
         // test single quoted package
-        let single_quoted_pkg = String::from("'foobar >= 1.0'");
-        let pkgs: Vec<String> = split_whitespace_unless_quoted(&single_quoted_pkg).collect();
+        let single_quoted_pkg = "'foobar >= 1.0'";
+        let pkgs: Vec<&str> = split_whitespace_unless_quoted(&single_quoted_pkg)?.collect();
```

Comment:
This is completely fine as is, but you can also just use:
`let pkgs: Vec<_> = ...`
Here the `.collect()` call needs to know that we're collecting into a `Vec<>`, but type inference can take over for there and know it must be `&str` because that's what `split_whitespace_unless_quoted()` returns.

This is totally fine as is though!

Reply:
Thanks :) Hopefully I'll get a hang of type inference and make use of it better soon. Still in "playing it safe" mode.

Reply:
Absolutely!  I'm writing this feedback not because there's any real problem with the code, but because it's a useful way to help learn!